Detectors/ITSMFT/common/workflow/src/DigitWriterSpec.cxx

```diff
@@ -41,15 +72,95 @@ DataProcessorSpec getDigitWriterSpec(bool mctruth, o2::header::DataOrigin detOri
   std::string detStr = o2::detectors::DetID::getName(detId);
   std::string detStrL = detStr;
   std::transform(detStrL.begin(), detStrL.end(), detStrL.begin(), ::tolower);
+#ifdef CUSTOM
+  std::vector<o2::framework::InputSpec> inputs;
+  if (mctruth) {
+    inputs.emplace_back(InputSpec{"digitsMCTR", detOrig, "DIGITSMCTR", 0});
+    inputs.emplace_back(InputSpec{"digitsMC2ROF", detOrig, "DIGITSMC2ROF", 0});
+  }
+  inputs.emplace_back(InputSpec{"digits", detOrig, "DIGITS", 0});
+  inputs.emplace_back(InputSpec{"digitsROF", detOrig, "DIGITSROF", 0});
+
+  return {(detStr + "DigitWriter").c_str(),
+          inputs,
+          {},
+          AlgorithmSpec{
+            [detStrL, detStr, mctruth](ProcessingContext& ctx) {
+              static bool mFinished = false;
+              if (mFinished) {
+                return;
+              }
+
+              TFile f((detStrL + "digits.root").c_str(), "RECREATE");
+              TTree t("o2sim", "o2sim");
+              // define data
+              auto digits = new std::vector<itsmft::Digit>; // needs to be a pointer since the message memory is managed by DPL and we have to avoid double deletes
+              auto rof = new std::vector<itsmft::ROFRecord>;
+              auto mc2rofrecords = new std::vector<itsmft::MC2ROFRecord>;
+              o2::dataformats::IOMCTruthContainerView labelview;
+
+              auto fillBranch = [](TBranch* br) {
+                br->Fill();
+                br->ResetAddress();
+                br->DropBaskets("all");
+              };
+              // get the data as gsl::span so that ideally no copy is made
+              // but immediately adopt the views in standard std::vectors *** THIS IS AN INTERNAL HACK ***
+              if (mctruth) {
+                labelview.adopt(ctx.inputs().get<gsl::span<char>>("digitsMCTR"));
+                fillBranch(t.Branch((detStr + "DigitMCTruth").c_str(), &labelview));
```

Comment:
This is a smart mechanism! I am just wondering if one should not the on exit "delete" these fake vectors by copying an empty `Imp` struct with all nullptrs? Not because of the memory leak of a few bytes but to not have lost dirty pointer?

Reply:
Yes absolutely. Thanks for pointing out.

Reply:
So the whole feature was moved into the RootTreeWriter. We could delete the custom treatment here.